strawberry/channels/testing.py

```diff
@@ -0,0 +1,102 @@
+import dataclasses
+import uuid
+from typing import AsyncIterator, Dict, List, Optional, Tuple, Union
+
+from graphql import GraphQLError
+
+from channels.testing.websocket import WebsocketCommunicator
+from strawberry.subscriptions import GRAPHQL_TRANSPORT_WS_PROTOCOL
+from strawberry.subscriptions.protocols.graphql_transport_ws.types import (
+    ConnectionAckMessage,
+    ConnectionInitMessage,
+    SubscribeMessage,
+    SubscribeMessagePayload,
+)
+from strawberry.types import ExecutionResult
+
+
+class GraphQLWebsocketCommunicator(WebsocketCommunicator):
+    """
+    Usage:
+    ```python
+    import pytest
+    from strawberry.channels.testing import GqlWsCommunicator
+    from myapp.asgi import application
+
+
+    @pytest.fixture
+    async def gql_communicator():
+        async with GqlWsCommunicator(application, path="/graphql") as client:
+            yield client
+
+
+    async def test_subscribe_echo(gql_communicator):
+        async for res in gql_communicator.subscribe(
+            query='subscription { echo(message: "Hi") }'
+        ):
+            assert res.data == {"echo": "Hi"}
+    ```
+    """
+
+    def __init__(
+        self,
+        application,
+        path: str,
+        headers: Optional[List[Tuple[bytes, bytes]]] = None,
+        subprotocols: Optional[List[str]] = None,
+    ):
+        subprotocols = (
+            subprotocols
+            if subprotocols
+            else [
+                GRAPHQL_TRANSPORT_WS_PROTOCOL,
+            ]
+        )
```

Comment:
Thinking about whether it would make sense to move the default subprotocols into the signature as the default. That way it would be more discoverable for users. What do you think?

Reply:
I think the old protocol is not supported by this communicator, I can try to resolve this though I don't think it worth the effort, this protocol is deprecated.

Reply:
I added support for graphql-ws.